File: src/main/java/com/swedbank/account/domian/model/Account.java
package com.swedbank.account.domian.model;

import com.swedbank.common.domian.EntityBase;
import com.swedbank.common.domian.Money;
import com.swedbank.user.domain.model.User;
import jakarta.persistence.Embedded;
import jakarta.persistence.Entity;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.Currency;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "accounts")
public class Account extends EntityBase {

    private String accountNumber;

    private String accountName;

    private Currency currency;

    @Embedded
    private Money balance;

    @ManyToOne
    private User user;
}

File: bank-frontend/src/app/core/models/account.ts
import { IBase } from '../../shared/models/base';
import { IMoney } from '../../shared/models/money';
import { IUser } from './user';

export interface IBankAccount extends IBase {
  accountNumber: string;
  accountName: string;
  balance: IMoney;
  user: IUser;
}
